### ISSUE TITLE
Re-classifying messages

### ISSUE THREAD

**Issue by henrikjohansen** [Graylog2/graylog2-server#449]
In an environment where stuff changes a lot it could be very beneficial to have the ability to re-classify messages.

One very real example is adding devices that already have existing messages in ES to a stream. Another could be the merger of 2 separate streams. A third possibility is the ability to re-run messages through an recently added extractor. 

I feel that this is a necessity for providing credibility and robustness to stream based analyses, reporting and alerting (since streams are such a vital anchor point for the way Graylog2 works).

The notion of long running system jobs already exists in Graylog2 - this would make a nice addon.

I am aware of complications, limitations, impacts, bla bla, because of the way ES works but this is rather essential because coherency and correctness are an absolute must for critical environments.

A fair compromise could be to expose this as an opt-in feature that requires clicking through 50 "I know I am stupid but I really want to do this anyways and nobody else but me is to blame" buttons and not doing this automagically upon events such as adding or modifying extractors).

**Comment by plcstpierre** [comment]
Just want to point out that this is still relevant today.

**Comment by mryanb** [comment]
Agreed with @plcstpierre  This is an advantage of Splunk, we're able to send data there and reclassify/index it down the road it we need to pull some more useful info down the road. It also allows us to ship the data there and worry about the reporting when its necessary.

**Comment by mriedmann** [comment]
Totally in need for this feature! 
In the meantime: Is there some alternative way to "rerun" old messages through new extractors?

**Comment by joschi** [comment]
@mriedmann You could read the messages from Elasticsearch with [Logstash and its "elasticsearch" input](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-elasticsearch.html) and send them again to Graylog with the [Logstash "gelf" output](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-gelf.html).

But be aware that this will essentially duplicate messages unless you remove the old ones from Elasticsearch.

**Comment by wjanowiak** [comment]
+1 for this feature

**Comment by thomaskneisel** [comment]
This feature would be soo nice :+1:

**Comment by panupan** [comment]
+1

**Comment by Dr-Crow** [comment]
Is there any update on this feature? It makes using Graylog less fluent.

**Comment by dsever** [comment]
+1

**Comment by fabboe** [comment]
+1

**Comment by rtntdeck** [comment]
+1

**Comment by lkarlslund** [comment]
+1 from here too :)

**Comment by joaociocca** [comment]
+1

**Comment by Sighawk** [comment]
+1

**Comment by DontPanic345** [comment]
+1

**Comment by K3UL** [comment]
+1

**Comment by habib-the-sweet** [comment]
+1

**Comment by lsadehaan** [comment]
+10

**Comment by dmz006** [comment]
+10

**Comment by Ollie42** [comment]
+10

**Comment by slavashvets** [comment]
+10

**Comment by PeterHurtenbach** [comment]
+10
